dashboard/app/serializers/script_seed_serializer.rb

```diff
@@ -0,0 +1,137 @@
+require 'digest'
+
+module ScriptSeed
+  def self.lesson_from_scope(scope, script_level)
+    lesson = scope[:lessons].select {|l| l.id == script_level.stage_id}.first
+    raise "No lesson found. scope: #{scope}" unless lesson
+    lesson
+  end
+
+  def self.lesson_group_from_scope(scope, script_level)
+    lesson = lesson_from_scope(scope, script_level)
+    lesson_group = scope[:lesson_groups].select {|lg| lg.id == lesson.lesson_group_id}.first
+    raise "No lesson group found. scope: #{scope}" unless lesson_group
+    lesson_group
+  end
+
+  def self.levels_from_scope(scope, script_level)
+    my_levels_script_levels = scope[:levels_script_levels].select {|lsl| lsl.script_level_id == script_level.id}
+    my_levels_script_levels.map do |lsl|
+      # n^2, probably doesn't matter since it's in memory and not that much stuff
+      level = scope[:levels].select {|l| l.id == lsl.level_id}.first
+      raise "No level found. scope: #{scope}" unless level
+      level
+    end
+  end
+
+  class LevelsScriptLevelSerializer < ActiveModel::Serializer
```

Comment:
I resisted this at first, because it seems like a bunch more code and state which could more compactly be stored as a list of level names on the script level. but I like that this follows the same pattern as the other tiers, and this may be "as simple as possible" i.e. trying to eliminate this tier from the serialized format may actually introduce more complexity. 

Let's keep this on the list of things to try to simplify, but if it's not possible then I'm on board with representing LevelsScriptLevel explicitly in the serialized format.

Reply:
As mentioned elsewhere, open to exploring this in a followup PR if it seems important, but it feels optional to me.